module/builder/collection/builder.go

```diff
@@ -60,277 +53,244 @@ func NewBuilder(
 		apply(&b.config)
 	}
 
-	return &b
+	// sanity check config
+	if b.config.ExpiryBuffer >= flow.DefaultTransactionExpiry {
```

Comment:
```suggestion
	if b.config.ExpiryBuffer > flow.DefaultTransactionExpiry {
```

I think it's OK to be equal, which is also consistent with the error message

Reply:
If they're equal, we would only allow transactions which have exactly our latest finalized block as the reference block. Practically it would mean very close to 0 transactions would be included. Maybe it's technically valid, but I think using `>=` is better in case there is a typo like `ExpiryBuffer=flow.DefaultTransactionExpiry`